### PR TITLE
bump up go-yaml library version

### PR DESCRIPTION
**Pull request by droot**
This PR updates the `go-yaml` version to latest version (Jan, 2021). The latest version of `go-yaml` has fixes related to roundtripping of comment in the YAML docs and it also introduces a formatting change (indentation) for items part of a sequence (list). Items in the list are indented by two spaces relative to parent (as with other types fields).

Refer to [commit](https://github.com/go-yaml/yaml/commit/a5ece683394c3b88d90572e44d36c93aea492c2c) for more details on the formatting and roundtripping fixes.

### CODE REVIEW

File: e2e/testdata/fn-render/generator/db/Kptfile
Comment (droot) [inline e2e/testdata/fn-render/generator/db/Kptfile:7]: I will update it to point to `gcr.io/kpt-fn/...` once it is there.

File: e2e/testdata/fn-render/resource-has-pkgname-prefix/Kptfile
Comment (droot) [inline e2e/testdata/fn-render/resource-has-pkgname-prefix/Kptfile:7]: I will update to the one from `gcr.io/kpt-fn/...` once it is there, using it for testing for now.

File: e2e/testdata/fn-render/resource-has-pkgname-prefix/wordpress-deployment.yaml
Comment (droot) [inline e2e/testdata/fn-render/resource-has-pkgname-prefix/wordpress-deployment.yaml:11]: We can see the roundtripping fixes working here in this example.

File: e2e/testdata/fn-render/resource-has-pkgname-prefix/.expected/diff.patch
Comment (droot) [inline e2e/testdata/fn-render/resource-has-pkgname-prefix/.expected/diff.patch:125]: roundtripping fixes in action here. Note `Copyright 2019 Google....` is no longer appearing here right next to the apiVersion field.

**Comment by mortent** [comment]
This looks good to me. But could we also include in the PR description the ways this change affects formatting? It looks like it only affects indentation for lists, but it would be great to have it listed.

**Comment by droot** [comment]
> This looks good to me. But could we also include in the PR description the ways this change affects formatting? It looks like it only affects indentation for lists, but it would be great to have it listed.

Updated the description and also added the link to the commit in `go-yaml` repo. Good thing, all e2e tests are also passing that are using newer version of all `kpt-functions` that are also updated to latest version of `go-yaml`. So this PR is good to go.

File: .golangci.yml
Comment (mortent) [inline .golangci.yml:34]: I think we should disable this rule in the specific places where we have violations. The syntax is:
```
//nolint:lll
```

**Comment by mortent** [comment]
I just found another issue in the current version of the yaml library that are fixed in the updated version. 
Marshalling of properties of type `yaml.Node` (as we are relying on [here](https://github.com/GoogleContainerTools/kpt/blob/722296a50a8d472a049efeed38f980852b290841/pkg/api/kptfile/v1alpha2/types.go#L242)) does not work with the current version of the library.

File: e2e/testdata/fn-render/generator/db/Kptfile
Comment (droot) [reply e2e/testdata/fn-render/generator/db/Kptfile]: Done.

File: e2e/testdata/fn-render/resource-has-pkgname-prefix/Kptfile
Comment (droot) [reply e2e/testdata/fn-render/resource-has-pkgname-prefix/Kptfile]: Done.

File: .golangci.yml
Comment (droot) [reply .golangci.yml]: Good tip: fixed it. Thanks

**Comment by droot** [comment]
> I just found another issue in the current version of the yaml library that are fixed in the updated version.
> Marshalling of properties of type `yaml.Node` (as we are relying on [here](https://github.com/GoogleContainerTools/kpt/blob/722296a50a8d472a049efeed38f980852b290841/pkg/api/kptfile/v1alpha2/types.go#L242)) does not work with the current version of the library.

That's great.

**Review by Shell32-Natsu — APPROVED** [review]
lgtm, but please wait for approval from @mortent

**Comment by Shell32-Natsu** [comment]
I assume all changes are test results update right?

**Review by mortent — APPROVED** [review]

**Comment by droot** [comment]
> I assume all changes are test results update right?

Yes.

**Pull request merged**
